tests/kuttl/tests/keystone_scale/05-assert.yaml

```diff
@@ -0,0 +1,52 @@
+#
+# Check for:
+#
+# - 0 KeystoneAPI CR
+# - 0 Pods for KeystoneAPI CR
+#
+
+apiVersion: keystone.openstack.org/v1beta1
+kind: KeystoneAPI
+metadata:
+  finalizers:
+  - KeystoneAPI
+  name: keystone
+  namespace: openstack
+spec:
+  adminProject: admin
+  adminRole: admin
+  adminUser: admin
+  containerImage: quay.io/tripleozedcentos9/openstack-keystone:current-tripleo
+  customServiceConfig: |
+    [DEFAULT]
+    debug = true
+  databaseInstance: openstack
+  databaseUser: keystone
+  debug:
+    bootstrap: false
+    dbSync: false
+    service: false
+  passwordSelectors:
+    admin: AdminPassword
+    database: KeystoneDatabasePassword
+  preserveJobs: false
+  region: regionOne
+  replicas: 0
+  resources:
+    requests:
+      cpu: "1.0"
+      memory: 500Mi
+  secret: osp-secret
+status:
+  apiEndpoint:
+    admin: http://keystone-admin-openstack.apps-crc.testing
+    internal: http://keystone-internal-openstack.apps-crc.testing
+    public: http://keystone-public-openstack.apps-crc.testing
+  databaseHostname: openstack
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: keystone
+spec:
+  replicas: 0
```

Comment:
I've tried adding it, but then the tests fail because the `availableReplicas` key is not found. I've done `oc get deployment keystone -o yaml` and the key is not there when the deployment is scaled down to 0 replicas, but it appears if the number of replicas are 1 or more

Reply:
yeah that is probably the infamous omitempty behavior of the k8s. The '0' value is the default / empty value of an integer field. And if a field is defined with the omitempty tag then k8s will not return the field if the value of the field is the empty value of the type of the field.

I'm not sure kuttl can taught that a missing field here is equal to the 0 value.  I opened an issue on kuttl to see if the devs there now a way to solve this. https://github.com/kudobuilder/kuttl/issues/455

Reply:
Based on the reply here https://github.com/kudobuilder/kuttl/issues/455#issuecomment-1377204077 we can sort of assert it by saying it is an error if there is such a field provided. I haven't tested this idea but it might work.

Reply:
I'm testing right now. It seems that we can use both an assert in the same test step. But the example errors does not seem to work. It seems to be looking for the string `{}` and not any value. So if I patch keystone to scale to 1, then it does not fail. I don't think kuttl allows partial/regex matching to check for a value that is not 0.
One solution would be to just check for a value of `availableReplicas` of 1, which is  the value from the previous test step. It seems brittle, but it might do the job, other than that, we would probably need some script to detect it.

Reply:
@cescgina thanks for trying out. Could you please bring back this to the original thread https://github.com/kudobuilder/kuttl/issues/455#issuecomment-1377204077 maybe the devs there has further ideas.

Reply:
Apparently there is no way to do it just with kuttl, I've pushed a commit with some scripting to do the check.